src/building/building_state.c

```diff
@@ -421,6 +423,10 @@ void building_state_load_from_buffer(buffer *buf, building *b, int building_buf_
     b->road_access_y = buffer_read_u8(buf);
     b->figure_id = buffer_read_i16(buf);
     b->figure_id2 = buffer_read_i16(buf);
+    if (save_version <= SAVE_GAME_CARAVANSERAI_LIGHTHOUSE_WRONG_ID && (b->type == BUILDING_LIGHTHOUSE || b->type == BUILDING_CARAVANSERAI)) {
+        b->figure_id = b->figure_id2;
+        b->figure_id2 = FIGURE_NONE;
```

Comment:
Even though this compiles correctly, `FIGURE_NONE` is used for the figure type, not the figure id. So this should be 0 instead.

Reply:
Whoops, fixed.